src/Microsoft.TestPlatform.Client/Execution/TestRunRequest.cs

```diff
@@ -522,6 +523,47 @@ public void HandleLogMessage(TestMessageLevel level, string message)
         /// <param name="rawMessage"></param>
         public void HandleRawMessage(string rawMessage)
         {
+            if (this.requestData.IsTelemetryOptedIn)
+            {
+                var message = this.dataSerializer.DeserializeMessage(rawMessage);
+
+                if (string.Equals(message.MessageType, MessageType.ExecutionComplete))
+                {
+                    var testRunCompletePayload =
+                        this.dataSerializer.DeserializePayload<TestRunCompletePayload>(message);
+
+                    if (testRunCompletePayload != null)
+                    {
+                        if (testRunCompletePayload.TestRunCompleteArgs?.Metrics == null)
+                        {
+                            testRunCompletePayload.TestRunCompleteArgs.Metrics = new Dictionary<string, object>();
```

Comment:
If it's null directly assign the `this.requestData.MetricsCollection.Metrics`, else loop through.

Reply:
fixed.